src/shared/install.c

```diff
@@ -588,7 +588,7 @@ static int create_symlink(
 
         if (symlink(old_path, new_path) >= 0) {
                 r = install_changes_add(changes, n_changes, INSTALL_CHANGE_SYMLINK, new_path, old_path);
-                if (r < 0)
+                if (r == -ENOMEM)
                         return r;
```

Comment:
so i think we should leave all calls as is that pass a positive 3rd argument. because in that case OOM is the only error the function can return. hence, drop this change and basically all cahnges where the 3rd argument isn't `r`